bee-tangle/src/ms_tangle.rs

```diff
@@ -0,0 +1,1145 @@
+        self.solid_entry_points.lock().await.insert(sep, index);
+    }
+
+    /// Removes the given solid entry point from the set of solid entry points.
+    pub async fn remove_solid_entry_point(&self, sep: &SolidEntryPoint) {
+        self.solid_entry_points.lock().await.remove(sep);
+    }
+
+    /// Clear all solid entry points.
+    pub async fn clear_solid_entry_points(&self) {
+        self.solid_entry_points.lock().await.clear();
+    }
+
+    /// Returns whether the message associated with given solid entry point is a solid entry point.
+    pub async fn is_solid_entry_point(&self, id: &MessageId) -> bool {
+        self.solid_entry_points
+            .lock()
+            .await
+            .contains_key(SolidEntryPoint::ref_cast(id))
+    }
+
+    /// Returns whether the message associated with the given message ID is solid.
+    pub async fn is_solid_message(&self, id: &MessageId) -> bool {
+        if self.is_solid_entry_point(id).await {
+            true
+        } else {
+            self.inner
+                .get_metadata(id)
+                .await
+                .map(|metadata| metadata.flags().is_solid())
+                .unwrap_or(false)
+        }
+    }
+
+    /// Get the oldest milestone root snapshot index.
+    pub async fn omrsi(&self, id: &MessageId) -> Option<IndexId> {
+        match self.solid_entry_points.lock().await.get(SolidEntryPoint::ref_cast(id)) {
+            Some(sep) => Some(IndexId::new(*sep, *id)),
+            None => match self.get_metadata(id).await {
+                Some(metadata) => metadata.omrsi(),
+                None => None,
+            },
+        }
+    }
+
+    /// Get the youngest milestone root snapshot index.
+    pub async fn ymrsi(&self, id: &MessageId) -> Option<IndexId> {
+        match self.solid_entry_points.lock().await.get(SolidEntryPoint::ref_cast(id)) {
+            Some(sep) => Some(IndexId::new(*sep, *id)),
+            None => match self.get_metadata(id).await {
+                Some(metadata) => metadata.ymrsi(),
+                None => None,
+            },
+        }
+    }
+
+    /// Insert the given message ID and parents as a tip.
+    pub async fn insert_tip(&self, message_id: MessageId, parents: Vec<MessageId>) {
+        self.tip_pool.lock().await.insert(&self, message_id, parents).await;
+    }
+
+    /// Update tip scores.
+    pub async fn update_tip_scores(&self) {
+        self.tip_pool.lock().await.update_scores(&self).await;
+    }
+
+    /// Return messages that require approving.
+    pub async fn get_messages_to_approve(&self) -> Option<Vec<MessageId>> {
+        self.tip_pool.lock().await.two_non_lazy_tips()
+    }
+
+    /// Reduce tips.
+    pub async fn reduce_tips(&self) {
+        self.tip_pool.lock().await.reduce_tips();
+    }
+
+    /// Return the number of non-lazy tips.
+    pub async fn non_lazy_tips_num(&self) -> usize {
+        self.tip_pool.lock().await.non_lazy_tips().len()
+    }
+}
+
+// #[cfg(test)]
```

Comment:
Is this intended to be commented?

Reply:
It's very old code, let's not keep it any longer.